noaastn/noaastn.py

```diff
@@ -1,42 +1,66 @@
 def get_stations_info(country="US", path=None):
-    """Downloads and cleans the data of all stations available at ftp://ftp.ncei.noaa.gov/pub/data/noaa/
+    """Downloads and cleans the data of all stations available at
+    'ftp://ftp.ncei.noaa.gov/pub/data/noaa/'.
 
     Parameters
     ----------
     country : str, optional
-              Filters station information by country location that is represented by two character country code or "all" for every country, by default "US"
+        Filters station information by country location that is represented by
+        two character country code or "all" for every country, by default "US".
     path : str, optional
-           path of the directory to save the data file. for example: "home/project/",
-           defaults to None that does not save the file.
+        Path of the directory to save the data file. For example:
+        "home/project/", defaults to None that does not save the file.
 
     Returns
     -------
     pandas.DataFrame
-        Data frame containing information of all stations
+        Data frame containing information of all stations.
 
     Examples
     --------
     >>> get_stations_info(country="US", path="home/project/")
     """
 
 
-def process_data(path=None):
+def get_weather_data(station_number, year, path=None):
     """
-    Extracts and cleans a time series of air temperature, atmospheric pressure,
-    wind speed, and wind direction from raw data file and returns a dataframe.
+    Download and clean weather data for a given NOAA station ID and year.
+    Returns a dataframe containing a time series of air temperature,
+    atmospheric pressure, wind speed, and wind direction.  Also saves a copy of
+    the raw data file downloaded from the NOAA FTP server at
+    'ftp.ncei.noaa.gov/pub/data/noaa/'.
 
     Parameters
     ----------
+    station_number : str
+        NOAA station number.
+    year : int
+        Year for which weather data should be returned
     path : str, optional
-           The path of the directory where raw data file is saved. Defaults
-           value (None) assumes that the raw data is in the root of the
-           repository.
+        The path of the directory where raw data file is saved. Default value
```

Comment:
Would it be nice if we avoid saving the data as a default option like the first function and do it if the user enters the path?
That we can avoid adding files to the user's machine if user does not wish to do so.

Reply:
Agreed - that would be ideal.  That being said, I haven't figured out how to implement the function yet without saving the file so I left it in at this point.  My thinking is we add the option not to save the file if we can figure out how to implement it.

Reply:
oh got it. Then we can go ahead with the current one.